### PR TITLE
Migrate from gulp

### PR DESCRIPTION
**Pull request by berkus**

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import { compilerOptions } from './tsconfig.json';
+
+// The entry point to the SDK.
+const input = ['src/index.ts'];
+
+// Destination folder.
+const distDir = 'dist';
+
+// Name of the SDK, will be reflected in the file name of the CDN version.
+const cdnSdkName = 'operation-retrier';
```
Comment (InfiniteRain) [inline rollup.config.js]: We probably don't need CDN versions of the bundle, as they're only ever consumed by our SDK's via NPM.

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      {
+        file: `${distDir}/browser.js`,
+        format: 'cjs',
+        exports: 'named',
+        banner: licenseBanner,
+      },
+    ],
+    external,
+  },
+  // Browser CDN bundle.
```
Comment (InfiniteRain) [inline rollup.config.js:109]: If previous comment is valid, then we shouldn't need the CDN bundle configs.

File: package.json [hunk 3]
```diff
@@ -19,7 +19,8 @@
     "build": "npx rollup -c",
     "sourceMapReport:html": "npx source-map-explorer@2.5.2 ./dist/lib.js --html ./dist/lib.report.html",
     "sourceMapReport:json": "npx source-map-explorer@2.5.2 ./dist/lib.js --json ./dist/lib.report.json",
-    "docs": "npx jsdoc@3.5.5 -c ./jsdoc.json"
+    "docs": "npx jsdoc@3.5.5 -c ./jsdoc.json",
+    "ci": "yarn clean && yarn lint && yarn test:unit && yarn build && yarn sourceMapReport:html && yarn sourceMapReport:json && yarn docs"
```
Comment (InfiniteRain) [inline package.json]: Didn't we decide to get rid of `yarn test:unit` part?

File: rollup.config.js (hunk 1 above)
Comment (berkus) [reply rollup.config.js]: kk

File: package.json (hunk 3 above)
Comment (berkus) [reply package.json]: Gone.

File: rollup.config.js (hunk 2 above)
Comment (berkus) [reply rollup.config.js:109]: Will keep for the bright future.

File: rollup.config.js (hunk 1 above)
Comment (berkus) [reply rollup.config.js]: Will keep for the bright future.

File: rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    resolve({
+      // Prefer builtin libraries. (E.G., when requiring 'events', it will
+      // prefer the node built-in version).
+      preferBuiltins: true,
+    }),
+    // Resolve commonjs syntax.
+    commonjs({
+      // Ignore these two modules as they're optional and the lib that requires
+      // them has fallbacks available.
+      ignore: ['bufferutil', 'utf-8-validate'],
```
Comment (InfiniteRain) [inline rollup.config.js]: this part not needed as twilsock is not a dependency in here

File: rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    // Resolve node modules.
+    resolve({
+      // Use browser resolution in imported modules.
+      browser: true,
+    }),
+    // Resolve commonjs syntax.
+    commonjs({
+      // Ignore these two modules as they're optional and the lib that requires
+      // them has fallbacks available.
+      ignore: ['bufferutil', 'utf-8-validate'],
```
Comment (InfiniteRain) [inline rollup.config.js]: this part not needed as twilsock is not a dependency in here

File: rollup.config.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  // Browser bundle.
+  {
+    ...browserConfig,
+    output: [
+      {
+        file: `${distDir}/browser.js`,
+        format: 'cjs',
+        exports: 'named',
+        banner: licenseBanner,
+      },
```
Comment (InfiniteRain) [inline rollup.config.js:105]: lets add `sourcemap: true` to here.

File: rollup.config.js (hunk 1 above)
Comment (InfiniteRain) [reply rollup.config.js]: ok but let's rename the variable to `bundleName` as this lib is not an sdk

**Review by InfiniteRain — APPROVED** [review]
lgtm

File: rollup.config.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+import { readFileSync } from 'fs';
+import { compilerOptions } from './tsconfig.json';
+
+// The entry point to the SDK.
+const input = ['src/index.ts'];
+
+// Destination folder.
+const distDir = 'dist';
+
+// Name of the SDK, will be reflected in the file name of the CDN version.
```
Comment (InfiniteRain) [inline rollup.config.js]: change "SDK" to "bundle" in the comment

**Review by InfiniteRain — CHANGES_REQUESTED** [review]

**Review by InfiniteRain — APPROVED** [review]

**Pull request merged**